module/platform/Webots/src/Webots.cpp

```diff
@@ -292,172 +304,206 @@ namespace module::platform {
+                        const int n                 = ::read(fd, initial_message.data(), sizeof(initial_message));
+
+                        if (n >= 0) {
+                            if (initial_message == "Welcome") {
+                                // good
+                                log<NUClear::INFO>(fmt::format("Connected to {}:{}", server_address, server_port));
+                            }
+                            else if (initial_message == "Refused") {
+                                log<NUClear::FATAL>(
+                                    fmt::format("Connection to {}:{} refused: your IP is not white listed.",
+                                                server_address,
+                                                server_port));
+                                // Halt and don't retry as reconnection is pointless.
+                                close(fd);
+                                powerplant.shutdown();
+                            }
+                            else {
+                                log<NUClear::FATAL>(
+                                    fmt::format("{}:{} sent unknown initial message", server_address, server_port));
+                                // Halt and don't retry as the other end is clearly not Webots
+                                close(fd);
+                                powerplant.shutdown();
+                            }
+                        }
+                        else {
+                            // There was nothing sent
+                            log<NUClear::DEBUG>("Connection was closed.");
+                            active_reconnect.store(false);
+                            return;
+                        }
+
+                        // Set the real time of the connection initiation
+                        connect_time = NUClear::clock::now();
+                        // Reset the simulation connection time
+                        utility::clock::last_update = NUClear::base_clock::now();
+
+                        connection_active = true;
+                    }
+                    else {
+                        // Work out how many bytes are available to read in the buffer and ensure we have enough
+                        // space to read them in our data buffer
+                        unsigned long available = 0;
+                        if (::ioctl(fd, FIONREAD, &available) < 0) {
+                            log<NUClear::ERROR>(fmt::format("Error querying for available data, {}", strerror(errno)));
+                            return;
+                        }
+                        const size_t old_size = buffer.size();
+                        buffer.resize(old_size + available);
+
+                        // Read data into our buffer and resize it to the new data we read
+                        auto bytes_read = ::read(fd, buffer.data() + old_size, available);
```

Comment:
I'm getting a warning about this being an unused variable. Does this need to be stored in a variable? And what's with the `::` here?

Reply:
The bytes_read variable should be used here? if it's not that's a problem

The `::` here is likely because it was stolen from some code I wrote for NUClear or similar. In that code I do this to make sure I'm getting the read function in POSIX, since I think there were other read functions floating around. The `::` at the start defines the root namespace, technically there is a `::` at the start of every single